api/services.go

```diff
@@ -31,6 +31,17 @@ func authHandler(sh *service.Handler, f endpointHandler) endpointHandler {
 			return f(s, r)
 		}
 
+		// Use certificate based authentication between cluster members.
+		if r.TLS != nil && r.Host == s.Address().URL.Host {
+			trustedCerts := s.Remotes().CertificatesNative()
+			for _, cert := range r.TLS.PeerCertificates {
+				trusted, _ := util.CheckTrustState(*cert, trustedCerts, nil, false)
+				if trusted {
+					return f(s, r)
+				}
+			}
+		}
+
```

Comment:
Yes, it would have failed before. Thars because before this PR, there was no reason for cluster member A to use the microcloud proxy to talk to the unix socket of a different service on cluster member B. 

The microcloud proxy is effectively only there for initializing/growing a cluster, so once the systems are set up, the proxy is not needed. In the growth case, the initiator is already set up, but the communication to joiners is uni-directional (as in the initiator reaches out to the joiners, never vice versa) so we didn't need to support the proxy to allow connections to the initiator.

However, with this PR we need to reach over the microcloud proxy to systems that are part of our microcloud cluster, but not the other services' clusters. So now we do have to consider and handle the case of using the microcloud proxy after the systems are initialized.

Reply:
Ok that makes sense, thanks :+1: